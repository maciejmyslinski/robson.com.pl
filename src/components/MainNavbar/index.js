import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid/v4';
import styled from 'styled-components';
import Link from 'gatsby-link';
import { Link as SimpleLink } from 'react-router-dom';

const Wrapper = styled.div`background-color: #424242;`;

const MainNavbar = props => {
  const { elements, env } = props;
  let LinkElement = env === 'gatsby' ? Link : SimpleLink;
  const Element = styled(LinkElement)`
    color: #fff;
    text-decoration: none;
    text-transform: uppercase;
    padding: 0 8px;
  `;
  return (
    <Wrapper>
      {elements.map(element =>
        <Element key={uuid()} to={element.path}>
          {element.name}
        </Element>
      )}
    </Wrapper>
  );
};

MainNavbar.propTypes = {
  elements: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      path: PropTypes.string
    })
  ).isRequired,
  env: PropTypes.oneOf(['gatsby', 'test'])
};

MainNavbar.defaultProps = {
  elements: [
    {
      name: 'Serwis',
      path: '/serwis'
    },
    {
      name: 'Sprężyny',
      path: '/sprezyny'
    },
    {
      name: 'Cenniki',
      path: '/cenniki'
    },
    {
      name: 'Kontakt',
      path: '/kontakt'
    }
  ],
  env: 'gatsby'
};

export default MainNavbar;
