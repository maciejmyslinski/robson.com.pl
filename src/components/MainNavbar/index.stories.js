import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { MemoryRouter } from 'react-router';
import MainNavbar from './';

storiesOf('MainNavbar', module).add('default', () => {
  return (
    <MemoryRouter>
      <MainNavbar env="test" />
    </MemoryRouter>
  );
});
