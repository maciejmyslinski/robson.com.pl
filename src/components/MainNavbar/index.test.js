import React from 'react';
import MainNavbar from './';

describe('MainNavbar', () => {
  it('should render', () => {
    const logo = {
      resolutions: {},
    };
    const component = shallow(<MainNavbar logo={logo} />);
    expect(component).toMatchSnapshot();
  });
});
