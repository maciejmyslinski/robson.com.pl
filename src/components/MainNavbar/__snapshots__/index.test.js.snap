// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavbar should render 1`] = `
<styled.div>
  <Styled(GatsbyLink)
    to="/serwis"
  >
    Serwis
  </Styled(GatsbyLink)>
  <Styled(GatsbyLink)
    to="/sprezyny"
  >
    Sprężyny
  </Styled(GatsbyLink)>
  <Styled(GatsbyLink)
    to="/cenniki"
  >
    Cenniki
  </Styled(GatsbyLink)>
  <Styled(GatsbyLink)
    to="/kontakt"
  >
    Kontakt
  </Styled(GatsbyLink)>
</styled.div>
`;
